#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Fix sem_trywait on amd64
# DP: Related bugs: 
# DP: Dpatch author: Clint Adams
# DP: Patch author: Upstream CVS
# DP: Upstream status: Committed
# DP: Status Details:
# DP: Date: 2005-12-16

PATCHLEVEL=0

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p$PATCHLEVEL < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p$PATCHLEVEL < $0;;
    *)
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.

--- nptl/sysdeps/unix/sysv/linux/x86_64/sem_trywait.S~	2003-05-10 16:37:38.000000000 -0400
+++ nptl/sysdeps/unix/sysv/linux/x86_64/sem_trywait.S	2005-10-24 16:50:36.000000000 -0400
@@ -1,4 +1,4 @@
-/* Copyright (C) 2002, 2003 Free Software Foundation, Inc.
+/* Copyright (C) 2002, 2003, 2005 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
    Contributed by Ulrich Drepper <drepper@redhat.com>, 2002.
 
@@ -56,4 +56,3 @@
 	orl	$-1, %eax
 	retq
 	.size	sem_trywait,.-sem_trywait
-	versioned_symbol(libpthread, __new_sem_trywait, sem_trywait, GLIBC_2_1)
